scripts/firedrake-configure

```diff
@@ -0,0 +1,433 @@
+    else:
+        assert args.show_env
+        print(" ".join(ENVIRONMENT_VARS[package_manager, arch]), end="")
+
+
+def sniff_package_manager() -> Optional[PackageManager]:
+    if platform.system() == "Linux":
+        if has_apt():
+            if platform.machine() == "x86_64":
+                return LINUX_APT_X86_64
+            elif platform.machine() == "aarch64":
+                return LINUX_APT_AARCH64
+    elif platform.system() == "Darwin":
+        if has_homebrew():
+            if platform.machine() == "arm64":
+                return MACOS_HOMEBREW_ARM64
+    raise RuntimeError(
+        "No package manager detected, if you want to configure Firedrake without "
+        "any system packages please pass '--no-package-manager'"
+    )
+
+
+def has_apt() -> bool:
+    try:
+        subprocess.run(["apt", "--version"], capture_output=True)
+        return True
+    except FileNotFoundError:
+        return False
+    except:  # noqa: E722
+        raise RuntimeError("Unexpected error occurred searching for apt")
+
+
+def has_homebrew() -> bool:
+    try:
+        subprocess.run(["brew", "--version"], capture_output=True)
+        return True
+    except FileNotFoundError:
+        return False
+    except:  # noqa: E722
+        raise RuntimeError("Unexpected error occurred searching for homebrew")
+
+
+BASE_LINUX_APT_PACKAGES = (
+    "bison",
+    "build-essential",
+    "cmake",
+    "flex",
+    "git",
+    "libopenblas-dev",
+    "libopenmpi-dev",
+    "ninja-build",
+    "pkg-config",
+    "python3-dev",
+    "python3-pip",
+)
+
+PETSC_EXTRAS_LINUX_APT_PACKAGES = (
+    "libfftw3-dev",
+    "libfftw3-mpi-dev",
+    "libhwloc-dev",
+    "libhdf5-mpi-dev",
+)
+
+COMMON_LINUX_APT_PACKAGES = BASE_LINUX_APT_PACKAGES + PETSC_EXTRAS_LINUX_APT_PACKAGES
+
+BASE_MACOS_HOMEBREW_PACKAGES = (
+    "autoconf",
+    "automake",
+    "boost",
+    "cmake",
+    "gcc",
+    "libtool",
+    "make",
+    "ninja",
+    "openblas",
+    "openmpi",
+    "pkg-config",
+    "python",
+)
+
+PETSC_EXTRAS_MACOS_HOMEBREW_PACKAGES = ("hwloc", "hdf5-mpi")
+
+COMMON_MACOS_HOMEBREW_PACKAGES = (
+    BASE_MACOS_HOMEBREW_PACKAGES + PETSC_EXTRAS_MACOS_HOMEBREW_PACKAGES
+)
+
+SYSTEM_PACKAGES = {
+    (LINUX_APT_X86_64, ARCH_DEFAULT): COMMON_LINUX_APT_PACKAGES,
+    (LINUX_APT_X86_64, ARCH_COMPLEX): COMMON_LINUX_APT_PACKAGES,
+    (LINUX_APT_AARCH64, ARCH_DEFAULT): COMMON_LINUX_APT_PACKAGES,
+    (LINUX_APT_AARCH64, ARCH_COMPLEX): COMMON_LINUX_APT_PACKAGES,
+    (MACOS_HOMEBREW_ARM64, ARCH_DEFAULT): COMMON_MACOS_HOMEBREW_PACKAGES,
+    (MACOS_HOMEBREW_ARM64, ARCH_COMPLEX): COMMON_MACOS_HOMEBREW_PACKAGES,
+}
+
+COMMON_PETSC_CONFIGURE_OPTIONS = (
+    "--with-c2html=0",
+    "--with-debugging=0",
+    "--with-fortran-bindings=0",
+    "--with-shared-libraries=1",
+    "--with-strict-petscerrorcode",
+    "--COPTFLAGS='-O3 -march=native -mtune=native'",
+    "--CXXOPTFLAGS='-O3 -march=native -mtune=native'",
+    "--FOPTFLAGS='-O3 -mtune=native'",
```

Comment:
Please can we only remove `-march=native` on macOS? MUMPS is Fortran so this is potentially an unecessary performance regression for our default solver for the majority of users.

Even better would be only removing `-march=native` on the apple clang compilers that don't recognise it, I think the newer apple clang has restored it. But I wouldn't hold up the PR for working out how to do that robustly (although I am biased as a non-apple user).

Reply:
Will do

Reply:
Thanks!

Reply:
done